### PR TITLE
feat(misconf): Add support for `--cf-params` for CFT

### PR DESCRIPTION
**Pull request by simar7**
## Description

Adds the support to pass in Cloudformation template files via `--cf-params` flag, similar to `-tf-vars`

## Related issues
- Fixes https://github.com/aquasecurity/trivy/issues/5167

## Required stacked PRs for release (this PR will be merged into the following PR)
- [x] https://github.com/aquasecurity/trivy/pull/5245/

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: docs/docs/coverage/iac/cloudformation.md [hunk 1]
```diff
@@ -2,23 +2,31 @@
...
-| :----: | :-------: |
+|:------:|:---------:|
 |  JSON  |     ✓     |
 |  YAML  |     ✓     |
 
 ## Misconfiguration
 Trivy recursively searches directories and scans all found CloudFormation files.
 It evaluates properties, functions, and other elements within CloudFormation files to detect misconfigurations.
 
+### Value Overrides
```
Comment (nikpivkin) [inline docs/docs/coverage/iac/cloudformation.md:20]: How about adding examples of supported formats or a link to aws-cli v2?

File: pkg/flag/misconf_flags.go [hunk 2]
```diff
@@ -55,6 +55,12 @@ var (
 		Default:    []string{},
 		Usage:      "specify paths to override the Terraform tfvars files",
 	}
+	CfParamsFlag = Flag{
+		Name:       "cf-params",
```
Comment (nikpivkin) [inline pkg/flag/misconf_flags.go:59]: CF scanner also has an option to pass parameters as key value (key1=value1,key2=value2). If we want to add support for this option in the future, will the choice of flag name cause problems?

Comment (simar7) [reply pkg/flag/misconf_flags.go:59]: I think it's cumbersome to pass key=value pairs via a command line option so I think it's fine for now. We don't plan to support such a case as it is error prone on user input / shell.

File: pkg/misconf/scanner.go [hunk 3]
```diff
@@ -115,6 +116,7 @@ func newScanner(t detection.FileType, filePatterns []string, opt ScannerOption)
 		scanner = arm.New(opts...)
 	case detection.FileTypeCloudFormation:
 		scanner = cfscanner.New(opts...)
+		configFiles = opt.CloudformationParamVars
```
[the remark this answers is not in the document]
Comment (simar7) [reply pkg/misconf/scanner.go]: Maybe I don't follow but this code is common to all scanners https://github.com/aquasecurity/trivy/blob/689f60a1b7352b01b8e1bbcb1aa7f684ea7aae80/pkg/misconf/scanner.go#L152-L151 - wouldn't we need to add support for all scanners before we can remove using config files to FS?

Can you elaborate?

File: docs/docs/coverage/iac/cloudformation.md (hunk 1 above)
Comment (simar7) [reply docs/docs/coverage/iac/cloudformation.md:20]: It's here https://github.com/aquasecurity/trivy/pull/5507/files#diff-e6c409fcf1992f2a1c44904cc9967f4d464c6fb3f8b30567d70f616812a6248dR32 which is referenced here https://github.com/aquasecurity/trivy/pull/5507/files#diff-e6c409fcf1992f2a1c44904cc9967f4d464c6fb3f8b30567d70f616812a6248dR21

Comment (nikpivkin) [reply docs/docs/coverage/iac/cloudformation.md:20]: [This](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html) documentation describes the structure of template parameters, but not the format of parameters defined in the file.

File: pkg/misconf/scanner.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nikpivkin) [reply pkg/misconf/scanner.go]: The option to pass FS with configuration files only exists for terraform and CloudFormation scanners, the others don't need it. I thought we can use this option in this PR for CloudFormation scanner, and in another PR use it for terraform scanner and remove the hack with adding config files.

[the remark this answers is not in the document]
Comment (simar7) [reply pkg/misconf/scanner.go]: Ah I see what you mean. I fixed it here https://github.com/aquasecurity/trivy/pull/5507/commits/8d27d95d16b13c746b61ced87100d093f9f40744

File: docs/docs/coverage/iac/cloudformation.md (hunk 1 above)
Comment (simar7) [reply docs/docs/coverage/iac/cloudformation.md:20]: Added https://github.com/aquasecurity/trivy/pull/5507/commits/8d27d95d16b13c746b61ced87100d093f9f40744

File: pkg/misconf/scanner.go [hunk 4]
```diff
@@ -114,6 +115,7 @@ func newScanner(t detection.FileType, filePatterns []string, opt ScannerOption)
 	case detection.FileTypeAzureARM:
 		scanner = arm.New(opts...)
 	case detection.FileTypeCloudFormation:
+		opts = append(opts, cfscanner.WithParameterFiles(opt.CloudFormationParamVars...))
```
Comment (nikpivkin) [inline pkg/misconf/scanner.go:118]: The configuration files must be added to a separate FS using the `WithConfigFS` option. Or write to the current one via `configFiles = opt.CloudFormationParamVars`

**Comment by simar7** [comment]
@knqyf263 could you take a look at reviewing it?

File: docs/docs/coverage/iac/cloudformation.md [hunk 5]
```diff
@@ -2,23 +2,34 @@
...
+You can find an example of CloudFormation parameters [here](CloudFormation Params Example)
+
 ## Secret
 The secret scan is performed on plain text files, with no special treatment for CloudFormation.
 
 [Misconfiguration]: ../../scanner/misconfiguration/index.md
-[Secret]: ../../scanner/secret.md
\ No newline at end of file
+[Secret]: ../../scanner/secret.md
+[CloudFormation Params]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html
```
Comment (knqyf263) [inline docs/docs/coverage/iac/cloudformation.md:34]:
```suggestion
[CloudFormation Parameters]: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html
```

File: docs/docs/references/configuration/cli/trivy_image.md [hunk 6]
```diff
@@ -36,6 +36,7 @@ trivy image [flags] IMAGE_NAME
 ```
       --cache-backend string              cache backend (e.g. redis://localhost:6379) (default "fs")
       --cache-ttl duration                cache TTL when using redis as cache backend
+      --cf-params strings                 specify paths to override the Cloudformation parameters files
```
Comment (knqyf263) [inline docs/docs/references/configuration/cli/trivy_image.md:39]: We may want to disable the flag for image scanning.

File: docs/docs/references/configuration/cli/trivy_kubernetes.md [hunk 7]
```diff
@@ -30,6 +30,7 @@ trivy kubernetes [flags] { cluster | all | specific resources like kubectl. eg:
   -A, --all-namespaces                    fetch resources from all cluster namespaces
       --cache-backend string              cache backend (e.g. redis://localhost:6379) (default "fs")
       --cache-ttl duration                cache TTL when using redis as cache backend
+      --cf-params strings                 specify paths to override the Cloudformation parameters files
```
Comment (knqyf263) [inline docs/docs/references/configuration/cli/trivy_kubernetes.md:33]: ditto

File: docs/docs/references/configuration/cli/trivy_vm.md [hunk 8]
```diff
@@ -23,6 +23,7 @@ trivy vm [flags] VM_IMAGE
       --aws-region string                 AWS region to scan
       --cache-backend string              cache backend (e.g. redis://localhost:6379) (default "fs")
       --cache-ttl duration                cache TTL when using redis as cache backend
+      --cf-params strings                 specify paths to override the Cloudformation parameters files
```
Comment (knqyf263) [inline docs/docs/references/configuration/cli/trivy_vm.md:26]: same here

File: pkg/flag/misconf_flags.go [hunk 9]
```diff
@@ -55,6 +55,12 @@ var (
...
+		ConfigName: "misconfiguration.cloudformation.vars",
```
Comment (knqyf263) [inline pkg/flag/misconf_flags.go:60]: nit: `cloudformation.vars` is a bit different from the CLI flag. Is it intended? `cloudformation.params` seems more aligned.

**Review by knqyf263 — COMMENTED** [review]
Looks great! I left small comments.

File: docs/docs/references/configuration/cli/trivy_image.md (hunk 6 above)
Comment (simar7) [reply docs/docs/references/configuration/cli/trivy_image.md]: that's right, I updated it here: https://github.com/aquasecurity/trivy/pull/5507/commits/8d8d834bc6c279a51e5c6962ad280502302587d9

**Review by knqyf263 — APPROVED** [review]
LGTM, but I'd defer to @nikpivkin. He recently spent more time on misconfiguration scanning than me.

**Review by nikpivkin — APPROVED** [review]
LGTM

**Pull request merged**